feat: allow rows without headers

defuncart opened the pull request:
This PR adds a couple of features which are useful on desktop:
- Allow rows without headers
- Allow rows to be highlighed/selected

Also added a theme for the widget.

![Bildschirmfoto 2022-11-10 um 13 12 36](https://user-images.githubusercontent.com/13286425/201089138-c1785c69-7faa-4096-ae9a-4592d61f6e96.png)

Highlighting and selected works well when there is no row header. In the case of row header, this is disabled.

defuncart committed:
feat: allow rows without headers

defuncart committed:
feat: allow rows to be selected

defuncart committed:
feat: Add DataGridThemeData

ted-marozzi commented:
![image](https://user-images.githubusercontent.com/38032037/201451395-5aa657db-665e-41d6-8fc8-332f77d0e81c.png)
Looks good, only thing is the column header is still frozen. Also could you document the highlighting issue with header rows so users will know. I'd also like to see what the issue is so we can try to fix it

defuncart committed:
feat: add documentation

defuncart committed:
feat: rows with frozen header highlight & select

defuncart commented:
> only thing is the column header is still frozen. 

I think this should be a separate feature, and is something I reckon most users would expect enabled by default. The `hasRowHeader` boolean is so that there are no frozen row headers.

> Also could you document the highlighting issue with header rows so users will know. 

Added some basic documentation, let me know if it could be improved.

> I'd also like to see what the issue is so we can try to fix it

Updated the code to cover both cases. This is achieved by syncing between `GridRowHeader` and `GridRows` which index is hovered.

ted-marozzi commented:
Sorry I wasn't clear, hasRowHeader is false, the cell at x,y 0,0 should not let the cell at x,y 0,1 go behind it. Do you disagree with that?

ted-marozzi commented:
Thanks for the docs, and great!

defuncart committed:
feat: hasRowHeader also applicable for column headers

defuncart commented:
> Sorry I wasn't clear, hasRowHeader is false, the cell at x,y 0,0 should not let the cell at x,y 0,1 go behind it. Do you disagree with that?

Sorry, missed that 🙈 Updated PR :)

defuncart commented:
While merging #12 and #14 on a local branch, there are some merge conflicts and I think it is better to merge #12 and then #14 into master, as the solution to #14 should be updated once #12 is merged. Thus marking #14 as draft.
